src/Chapter3.hs

```diff
@@ -971,6 +1008,20 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+data Week = Mon | Tue | Wed | Thu | Fri | Sat | Sun
+    deriving (Show, Enum, Eq, Ord, Bounded)
+
+isWeekend :: Week -> Bool
+isWeekend day = day == Sat || day == Sun
+
+nextDay :: Week -> Week
+nextDay day
+    | day == maxBound = minBound
+    | otherwise = succ day
+
+daysToParty :: Week -> Int
+daysToParty day = abs (fromEnum day - fromEnum Fri)
```

Comment:
Unfortunately, time doesn't work this way, and we can't travel back in time disappointed
For example, if today is Saturday, there are 6 days to the next Friday, not 1 (as in your current solution).